pkg/node/node.go

```diff
@@ -256,13 +260,19 @@ func (n *Node) monitorKeepPublicKeySubmission(
 			event.SubmittingMember,
 			event.ConflictingPublicKey,
 		)
-	case <-ctx.Done():
-		if ctx.Err() == context.DeadlineExceeded {
+	case <-monitoringCtx.Done():
+		if monitoringCtx.Err() == context.DeadlineExceeded {
```

Comment:
Do we need this check? When the context gets cancelled we always have a timeout in this case as far as I understand.

Reply:
It can get canceled also by calling context cancel function.